### PR TITLE
Vendor and smartfridge cleanup.

### PR DESCRIPTION
**Pull request by MistakeNot4892**
## Description of changes
This should possibly be going to dev, but despite the footprint, is not really a significant code change.
- Breaks up vending.dmi.
- Moves smartfridges into their own folder.
- Makes smartfridges and vendors use add_overlay().
- Cleans up smartfridge/vendor icon updates.
- Adds required states for bar fridges, drying ovens and nutrimax.
- Adds unit tests for vendor and smartfridge icons/overlays.

## Why and what will this PR improve
- Fixes https://github.com/ScavStation/ScavStation/issues/1127
- Tests vendors and smartfridges in CI.
- Conceptually separates vendors and smartfridges.
- Allows easily moving vendor/smartfridge icons into modpacks.

## Authorship
Myself.

## Changelog
Nothing player-facing.

### CODE REVIEW

File: code/game/machinery/smartfridge/foods.dm [hunk 1]
```diff
@@ -0,0 +1,9 @@
+/obj/machinery/smartfridge/foods
+	name = "\improper Hot Foods Display"
+	desc = "A heated storage unit for piping hot meals."
+	icon = 'icons/obj/machines/smartfridges/food.dmi'
+	overlay_contents_icon = 'icons/obj/machines/smartfridges/contents_food.dmi'
+
+/obj/machinery/smartfridge/foods/accept_check(var/obj/item/O)
+	if(istype(O,/obj/item/food) || istype(O,/obj/item/utensil))
```
Comment (out-of-phaze) [inline code/game/machinery/smartfridge/foods.dm]: should maybe add soup bowls to this? could even check that they have reagents in them

File: code/game/machinery/vending/_vending.dm [hunk 2]
```diff
@@ -4,8 +4,8 @@
 /obj/machinery/vending
 	name = "Vendomat"
 	desc = "A generic vending machine."
-	icon = 'icons/obj/vending.dmi'
-	icon_state = "generic"
+	icon = 'icons/obj/machines/vending/generic.dmi'
+	icon_state = ICON_STATE_WORLD
```
Comment (out-of-phaze) [inline code/game/machinery/vending/_vending.dm:8]: I really don't get the point of using `ICON_STATE_WORLD` for non-objects. An inventory or worn sprite is never valid here. It's not like vendors are the only things that do this, obviously, but still, I don't rrrrrrrrreally get it?

File: code/unit_tests/time_tests.dm [hunk 3]
```diff
@@ -51,7 +51,7 @@
 	return TRUE
 
 /datum/unit_test/time/shall_not_validate_range_that_exlude_sixt_of_june_start_before_end
-	name = "Shall not validate range that exlude 6th of June - Start before End"
+	name = "TIME: Shall not validate range that exlude 6th of June - Start before End"
```
Comment (out-of-phaze) [inline code/unit_tests/time_tests.dm]: while you're at it there's a typo here and on line 66, `exlude` :)

**Review by out-of-phaze — DISMISSED** [review]
fwiw i'm pretty sure icondiffbot can handle a PR of this size just fine? i can double-check though

File: code/game/machinery/vending/_vending.dm (hunk 2 above)
Comment (MistakeNot4892) [reply code/game/machinery/vending/_vending.dm:8]: Main benefit is not having to set icon_state on subtypes, using `world` just tickles the consistency part of my brain.

File: code/game/machinery/smartfridge/foods.dm [hunk 4]
```diff
@@ -5,5 +5,10 @@
...
 /obj/machinery/smartfridge/foods/accept_check(var/obj/item/O)
-	if(istype(O,/obj/item/food) || istype(O,/obj/item/utensil))
-		return 1
+	var/static/list/_food_types = list(
+		/obj/item/food,
+		/obj/item/utensil,
+		/obj/item/chems/glass/bowl,
+		/obj/item/chems/glass/handmade/bowl
+	)
+	return istype(O) && O.reagents?.total_volume && is_type_in_list(O, _food_types)
```
Comment (out-of-phaze) [inline code/game/machinery/smartfridge/foods.dm:14]: this requires that utensils have reagents in them

**Review by out-of-phaze — CHANGES_REQUESTED** [review]

**Review by out-of-phaze — APPROVED** [review]

File: code/game/machinery/smartfridge/foods.dm (hunk 4 above)
Comment (MistakeNot4892) [reply code/game/machinery/smartfridge/foods.dm:14]: I don't really know why the hot foods fridge needs to take utensils, honestly.

**Pull request merged**